### PR TITLE
Fixing Code Quality Issues

### PR DESCRIPTION
**Pull request by HarshCasper**
### Description
Hi 👋 

I ran the DeepSource static analyzer on the forked copy of this repo and found some interesting code quality [issues](https://deepsource.io/gh/HarshCasper/tailscale/). This PR fixes a few of them.

### Summary of changes

- Combine multiple `append`s into a single call
- Fix check for empty string
- Remove unnecessary use of slice
- Added `.deepsource.toml` file

### CODE REVIEW

**Comment by bradfitz** [comment]
None of these look like actual quality issues. These are just arbitrary style changes. The new code loses the explicitness of the old.

**Comment by bradfitz** [comment]
I'm okay with some of these changes if you're open to doing edits on this PR. Are you?

If so, I'll let you know which.

**Comment by HarshCasper** [comment]
> I'm okay with some of these changes if you're open to doing edits on this PR. Are you?
> 
> If so, I'll let you know which.

Sure. I would be more than happy to work on this :)

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version = 1
```
Comment (bradfitz) [inline .deepsource.toml:1]: Please remove this file. This isn't something we'll be running regularly.

File: disco/disco.go [hunk 2]
```diff
@@ -148,7 +148,7 @@ func (m *CallMeMaybe) AppendMarshal(b []byte) []byte {
 	ret, p := appendMsgHeader(b, TypeCallMeMaybe, v0, epLength*len(m.MyNumber))
 	for _, ipp := range m.MyNumber {
 		a := ipp.IP.As16()
-		copy(p[:], a[:])
+		copy(p, a[:])
```
Comment (bradfitz) [inline disco/disco.go:151]: Please revert. I prefer the explicitness.

File: net/packet/udp6.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ func (h UDP6Header) Marshal(buf []byte) error {
 
 	// UDP checksum with IP pseudo header.
 	h.IP6Header.marshalPseudo(buf)
-	binary.BigEndian.PutUint16(buf[46:48], ip4Checksum(buf[:]))
+	binary.BigEndian.PutUint16(buf[46:48], ip4Checksum(buf))
```
Comment (bradfitz) [inline net/packet/udp6.go:40]: Please revert. I prefer the explicitness.

File: tailcfg/tailcfg.go [hunk 4]
```diff
@@ -850,7 +850,7 @@ func keyUnmarshalText(dst []byte, prefix string, text []byte) error {
 	if err != nil {
 		return fmt.Errorf("UnmarshalText: after %q: %v", prefix, err)
 	}
-	copy(dst[:], pub[:])
+	copy(dst, pub[:])
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:853]: Please revert. I prefer the explicitness.

**Review by bradfitz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**